docs/deprecations.md

```diff
@@ -0,0 +1,207 @@
+
+### [`jsonserialisation.JSONSerialiser`](https://github.com/gchq/Gaffer/blob/gaffer2-1.21.1/core/serialisation/src/main/java/uk/gov/gchq/gaffer/jsonserialisation/JSONSerialiser.java)
+- The method `update(final String jsonSerialiserClass, final String jsonSerialiserModules)` has been removed.
+- Use `update(final String jsonSerialiserClass, final String jsonSerialiserModules, final Boolean strictJson)` instead. Passing `strictJson` as `null` will result in the same behaviour.
+
+### [`hdfs.operation.MapReduce`](https://github.com/gchq/Gaffer/blob/gaffer2-1.21.1/library/hdfs-library/src/main/java/uk/gov/gchq/gaffer/hdfs/operation/MapReduce.java)
+- The methods `getNumReduceTasks()` and `setNumReduceTasks(Integer)` have been removed.
+- Gaffer’s operations that inherit `MapReduce` did not make use of `numReduceTasks`, either setting it to a constant number in the `JobFactory` or using Accumulo to automatically set the number (recommended for performance) and using min/max to keep it within a range. Therefore, `numReduceTasks`, `getNumReduceTasks` and `setNumReduceTasks` have been removed from this interface.
+
+### [`hdfs.operation.AddElementsFromHdfs`](https://github.com/gchq/Gaffer/blob/gaffer2-1.21.1/library/hdfs-library/src/main/java/uk/gov/gchq/gaffer/hdfs/operation/AddElementsFromHdfs.java)
+- The methods `getNumReduceTasks()` and `setNumReduceTasks(Integer)` have been removed.
+- The number of reduce tasks should not be set.  By default the number of reduce tasks should match the number of tablets.  Use minimum and maximum reduce tasks to specify boundaries for the number of reduce tasks.
+
+### [`hdfs.operation.SampleDataForSplitPoints`](https://github.com/gchq/Gaffer/blob/gaffer2-1.21.1/library/hdfs-library/src/main/java/uk/gov/gchq/gaffer/hdfs/operation/SampleDataForSplitPoints.java)
+- The methods `getNumReduceTasks()` and `setNumReduceTasks(Integer)` have been removed.
+- These methods were not required as `NumReduceTasks` was always set to 1 in any case.
+	 
+### [`rest.SystemProperty`](https://github.com/gchq/Gaffer/blob/gaffer2-1.21.1/rest-api/common-rest/src/main/java/uk/gov/gchq/gaffer/rest/SystemProperty.java)
+- `GRAPH_ID`, `GRAPH_HOOKS_PATH`, `GRAPH_LIBRARY_PATH` and `GRAPH_LIBRARY_CONFIG` have been removed.
+- These config options have been removed in favour of providing a `graphConfig` JSON and using `GRAPH_CONFIG_PATH` instead.
+
+### [`rest.service.v2.example.ExamplesFactory`](https://github.com/gchq/Gaffer/blob/gaffer2-1.21.1/rest-api/core-rest/src/main/java/uk/gov/gchq/gaffer/rest/service/v2/example/ExamplesFactory.java)
+- This class has been removed.
+- It is replaced by `uk.gov.gchq.gaffer.rest.factory.ExamplesFactory`, which can be used instead.
+
+### [`accumulostore.AccumuloProperties`](https://github.com/gchq/Gaffer/blob/gaffer2-1.21.1/store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/AccumuloProperties.java)
```

Comment:
Not sure what order these are meant to be in but it's almost alphabetical. Perhaps first could be the Serialiser section, followed by a few key ones like SeedMatching and getTraits/hasTraits, then they go alphabetically? I am not too fussed just a thought

Reply:
I used the order of the deprecations in the generated list, not sure how that was ordered. I then moved a few around to ensure areas were not separated. It makes sense to do this alphabetically as it will preserve that grouping. I'll change the order as suggested.